File: ide/src/main/java/org/dau/ide/main/MainMenuBar.java
package org.dau.ide.main;

import javafx.scene.control.MenuBar;
import org.springframework.context.event.ContextRefreshedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

import static org.dau.ide.action.FxActions.fillMenuBar;

@Component
public final class MainMenuBar extends MenuBar {

  @EventListener
  public void onRefresh(ContextRefreshedEvent event) {
    fillMenuBar(event.getApplicationContext(), this, MainQualifier.class);
  }
}

File: ide/src/main/kotlin/org/dau/ide/main/MainMenuBar.kt
package org.dau.ide.main

import javafx.scene.control.MenuBar
import org.dau.ide.action.FxAction.fillMenuBar
import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class MainMenuBar : MenuBar() {

  @EventListener
  fun onRefresh(event: ContextRefreshedEvent) {
    fillMenuBar(event.applicationContext, this, MainQualifier::class.java)
  }
}
